### PR TITLE
fix(developer): enchancer for adding is_dev_account to tracking events

### PR DESCRIPTION
**Pull request by nightnei**
Related to https://github.com/Automattic/dotcom-forge/issues/5456

## Proposed Changes
Enchancer of track events by is_dev_account for "Learn more" links on developer/me page.

## Testing Instructions
1. Open `http://calypso.localhost:3000/me/developer`
2. Click on any "Learn more" link or link inside description
3. Assert that appropriate `is_dev_account` is added to event

### CODE REVIEW

**Comment by wongasy** [comment]
@nightnei Thanks for the quick workaround! Indeed this works as expected (FYI, I've also rebased from trunk)! I found other places in the code where a similar approach was taken: https://github.com/Automattic/wp-calypso/blob/7219704d07fda7215708b539dc2f693138142df8/client/landing/subscriptions/tracks/use-record-subscriptions-tracks-event.tsx#L6 

What do you think about creating a hook `useRecordTracksEventWithUserIsDevAccount()` so that the `user_is_dev_account` prop can be applied to any tracks event?

**Comment by nightnei** [comment]
@wongasy Do you mean https://github.com/Automattic/wp-calypso/pull/87100/files#diff-f37180b64e56894728b1d8f5cc83971d7d858e22eaea6f74e074ac23c4e1eee3R1-R19?
If so - I am good with it.

File: client/me/developer/features/use-record-tracks-event-with-user-is-dev-account.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { recordTracksEvent, enhanceWithUserIsDevAccount } from 'calypso/state/analytics/actions';
+import { withEnhancers } from 'calypso/state/utils';
+import type { AnyAction, Store } from 'redux';
+import type { ThunkDispatch } from 'redux-thunk';
+
+export const useRecordTracksEventWithUserIsDevAccount = () => {
```
Comment (wongasy) [inline client/me/developer/features/use-record-tracks-event-with-user-is-dev-account.tsx:16]:
```suggestion
/**
 * Get a function that enhances any tracks event with an additional property `user_is_dev_account`
 * to determine if the event was triggered by a user with the `is_dev_account` setting enabled.
 *
 * Example:
 * const recordTracksEventWithUserIsDevAccount = useRecordTracksEventWithUserIsDevAccount();
* <MyComponent onClick={ () => { recordTracksEventWithUserIsDevAccount( 'some_event', { 'data': 'test' } ) } >
 */
export const useRecordTracksEventWithUserIsDevAccount = () => {
```

Let's document what this function does.

**Review by wongasy — APPROVED** [review]
LGTM! I just left one non-blocking feedback.

Confirming that tracks event for the "Learn more" button now includes the `user_is_dev_account` property:

Screen Shot 2024-02-02 at 11 32 58 AM

**Pull request merged**